### DISCUSSION TITLE
Clear step by step instructions for 5.3.201 and Visual Studio on how to identify the code to modify, to test the change and to propose that change when OK

### DISCUSSION THREAD

**Discussion by dobo365** [RomanKubiak/ctrlr#99] [Q&A]
Hi,
  I would gladly support the development/improvement of Ctrlr with simple things.
I'm using 5.3.201 and at this stage do not plan to use the latest version (of course this may change).
I'm not familiar enough with the usage of different versions and Visual Studio and thus would it be possible to get a clear step by step instructions for 5.3.201 and Visual Studio on how to identify the code to modify, to test the change and to propose that change when OK?
The things I'd like to improve/contribute to:
- help text for the properties: there is a link foreseen but doesn't display anything
- tab management: deleting a tab doesn't delete the child modulators, moving a tab doesn't move the child components...
- a few other small issues
Thx in advance!

**Comment by Ted-juh** [comment]
Is this helpful:

https://ctrlr.org/forums/topic/build-ctrlr-on-windows-10-manual/

Sorry Goodweather, just read in another discussion that you are working on Linux. My manual is for Windows 10 and is probably not suitable for compiling Ctrlr on Linux. Sorry for the misunderstanding.

**Reply by dobo365** [reply]
No, I'm not working on Linux. It was just a suggestion of what could be possible to enhance once we know how to build Ctrlr.

**Comment by dobo365** [comment]
Heel bedankt Tedjuh!
Thx a lot for the link, I was not aware.
In fact wanted to contact you to have some Skype/Zoom meeting together. Maybe we can still do that ;-)
Question: why using VS2010 and not the latest? What is the issue?

I will look at the doc and try to understand/follow what you did.

**Reply by Ted-juh** [reply]
Why VS2010? I was aiming to compile what should be, in my opinion, Ctrlr version 5.3.201. There is a sln (solution) file in that commit on Github that is for VS2010. So I figured I better use that version. 

To be honest, I didn't bother to try it with other versions of VS. When you try to open a VS2010 sln file with a newer version of VS it does some re-linking or rebuilding of some files and that might cause some trouble when compiling because it can't find certain files that are needed to compile CTRLR. Mostly because newer versions of VS are using newer versions of MSBuild and other files it depends on. So, it might work, it might not. It might rain errors like a normal day in the UK or it's a sunny happy walk through the park in Vienna. 

Hit me up on "tedjuh at-symbol yahoo.com" if you need VS2010 and/ or want to make an appointment. :)

**Reply by dobo365** [reply]
OK!
I downloaded Visual Studio 2010 Ultimate which was a demo version then found a product key to register it.
It is now in parallel to my VS 2019.
Tomorrow or in the coming days I will follow your step by step and report. Thx a lot for providing this. I will for sure learn and complete my knowledge.
If we get it working on 2010, we should try to get it working on VS2019 just to be up to date in terms of dev tool.
But this is another story. Let me first walk before I start running ;-)
